File: Database/Assembly_SRPG_JP/VersusEnableTimeParam.cs
// Decompiled with JetBrains decompiler
// Type: SRPG.VersusEnableTimeParam
// Assembly: Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 85BFDF7F-5712-4D45-9CD6-3465C703DFDF
// Assembly location: S:\Desktop\Assembly-CSharp.dll

using System;
using System.Collections.Generic;

namespace SRPG
{
  public class VersusEnableTimeParam
  {
    private int mScheduleId;
    private VERSUS_TYPE mVersusType;
    private DateTime mBeginAt;
    private DateTime mEndAt;
    private List<VersusEnableTimeScheduleParam> mSchedule;
    private VersusDraftType mDraftType;

    public int ScheduleId
    {
      get
      {
        return this.mScheduleId;
      }
    }

    public VERSUS_TYPE VersusType
    {
      get
      {
        return this.mVersusType;
      }
    }

    public DateTime BeginAt
    {
      get
      {
        return this.mBeginAt;
      }
    }

    public DateTime EndAt
    {
      get
      {
        return this.mEndAt;
      }
    }

    public List<VersusEnableTimeScheduleParam> Schedule
    {
      get
      {
        return this.mSchedule;
      }
    }

    public VersusDraftType DraftType
    {
      get
      {
        return this.mDraftType;
      }
    }

    public bool Deserialize(JSON_VersusEnableTimeParam json)
    {
      if (json == null)
        return false;
      this.mScheduleId = json.id;
      this.mVersusType = (VERSUS_TYPE) json.mode;
      try
      {
        if (!string.IsNullOrEmpty(json.begin_at))
          this.mBeginAt = DateTime.Parse(json.begin_at);
        if (!string.IsNullOrEmpty(json.end_at))
          this.mEndAt = DateTime.Parse(json.end_at);
      }
      catch (Exception ex)
      {
        DebugUtility.LogError(ex.Message);
        return false;
      }
      this.mSchedule = new List<VersusEnableTimeScheduleParam>();
      for (int index = 0; index < json.schedule.Length; ++index)
      {
        VersusEnableTimeScheduleParam timeScheduleParam = new VersusEnableTimeScheduleParam();
        if (timeScheduleParam.Deserialize(json.schedule[index]))
          this.mSchedule.Add(timeScheduleParam);
      }
      this.mDraftType = (VersusDraftType) json.draft_type;
      return true;
    }
  }
}

File: Database_V2/ParamFunctions_untested/VersusEnableTimeParam.py
def VersusEnableTimeParam(json):
    this={}#VersusEnableTimeParamjson)
    #if(json==null)
    #returnfalse
    if 'id' in json:
        this['mScheduleId'] = json['id']
    if 'mode' in json:
        this['mVersusType'] = ENUM['VERSUS_TYPE'][json['mode']]
    #try
        #if(!string.IsNullOrEmpty(json.begin_at))
        if 'begin_at' in json:
            this['mBeginAt'] = DateTime.Parse
        #if(!string.IsNullOrEmpty(json.end_at))
        if 'end_at' in json:
            this['mEndAt'] = DateTime.Parse
    #catch(Exceptionex)
        #DebugUtility.LogError(ex.Message)
        #returnfalse
    #this.mSchedule=newList<VersusEnableTimeScheduleParam>()
    #for(intindex=0index<json.schedule.Length++index)
        #VersusEnableTimeScheduleParamtimeScheduleParam=newVersusEnableTimeScheduleParam()
        #if(timeScheduleParam.Deserialize(json.schedule))
        #this.mSchedule.Add(timeScheduleParam)
    if 'draft_type' in json:
        this['mDraftType'] = ENUM['VersusDraftType'][json['draft_type']]
    #returntrue
return this
